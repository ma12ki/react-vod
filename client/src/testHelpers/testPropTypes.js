import React from 'react';

// this has some limitations:
// https://stackoverflow.com/questions/41916992/why-does-my-jest-test-of-react-proptypes-break-when-using-multiple-unacceptable/41916993#41916993
export const testPropTypes = (component, propName, acceptableValues, unacceptableValues, otherProps) => {
    console.error = jest.fn();
    const _test = (testValues, expectError) => {
        for (let propValue of testValues) {
            console.error.mockClear();
            React.createElement(component, {...otherProps, [propName]: propValue});
            try {
                expect(console.error).toHaveBeenCalledTimes(expectError ? 1 : 0);
            } catch (err) {
                console.log(`${propName} expected to ${expectError ? '' : 'NOT'} fail for value ${propValue}`);
                throw err;
            }
        }
    };
    _test(acceptableValues || [], false);
    _test(unacceptableValues || [], true);
};
