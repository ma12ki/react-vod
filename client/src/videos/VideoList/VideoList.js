import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Table } from 'reactstrap';

import { getFilteredVideos } from '../selectors';
import { VideoItem } from '../VideoItem';
import { utils } from '../../shared/cssModules';

export const VideoList = ({ items }) => {
    const videos = items.map((video) => {
        return (
            <VideoItem key={video.id} video={video} />
        );
    })

    return (
        <Table>
            <thead>
                <tr>
                    <th className={utils.alignLeft}>Name</th>
                    <th className={utils.center}>Date created</th>
                    <th className={utils.alignRight}>Duration</th>
                    <th className={utils.alignRight}>Size</th>
                    <th></th>
                </tr>
            </thead>
            <tbody>
                {videos}
            </tbody>
        </Table>
    );
};

VideoList.propTypes = {
    items: PropTypes.array.isRequired
};

const mapStateToProps = (state) => ({
    items: getFilteredVideos(state),
});

export default connect(mapStateToProps)(VideoList);
