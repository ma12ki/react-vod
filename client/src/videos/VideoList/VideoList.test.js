import React from 'react';
import { shallow } from 'enzyme';

import { testPropTypes } from '../../testHelpers';
import { VideoList } from './VideoList';

const makeWrapper = ({ items = [] } = {}) => shallow(<VideoList items={items} />);

describe('VideoList', () => {

    it('matches the snapshot', () => {
        const wrapper = makeWrapper();

        expect(wrapper).toMatchSnapshot();
    });

    it('requires "items" object prop', () => {
        testPropTypes(
            VideoList,
            'items',
            [[]],
            [123, '123', null, undefined]
        );
    });

});