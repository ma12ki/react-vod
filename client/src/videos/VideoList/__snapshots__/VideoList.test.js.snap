// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoList matches the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Table
    responsiveTag="div"
    tag="table"
>
    <thead>
        <tr>
            <th
                className="alignLeft"
            >
                Name
            </th>
            <th
                className="center"
            >
                Date created
            </th>
            <th
                className="alignRight"
            >
                Duration
            </th>
            <th
                className="alignRight"
            >
                Size
            </th>
            <th />
        </tr>
    </thead>
    <tbody>
        
    </tbody>
</Table>,
  "nodes": Array [
    <Table
      responsiveTag="div"
      tag="table"
>
      <thead>
            <tr>
                  <th
                        className="alignLeft"
                  >
                        Name
                  </th>
                  <th
                        className="center"
                  >
                        Date created
                  </th>
                  <th
                        className="alignRight"
                  >
                        Duration
                  </th>
                  <th
                        className="alignRight"
                  >
                        Size
                  </th>
                  <th />
            </tr>
      </thead>
      <tbody>
            
      </tbody>
</Table>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown
        items={Array []}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "items": Array [],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Table
          responsiveTag="div"
          tag="table"
>
          <thead>
                    <tr>
                              <th
                                        className="alignLeft"
                              >
                                        Name
                              </th>
                              <th
                                        className="center"
                              >
                                        Date created
                              </th>
                              <th
                                        className="alignRight"
                              >
                                        Duration
                              </th>
                              <th
                                        className="alignRight"
                              >
                                        Size
                              </th>
                              <th />
                    </tr>
          </thead>
          <tbody>
                    
          </tbody>
</Table>,
        "_debugID": 2,
        "_renderedOutput": <Table
          responsiveTag="div"
          tag="table"
>
          <thead>
                    <tr>
                              <th
                                        className="alignLeft"
                              >
                                        Name
                              </th>
                              <th
                                        className="center"
                              >
                                        Date created
                              </th>
                              <th
                                        className="alignRight"
                              >
                                        Duration
                              </th>
                              <th
                                        className="alignRight"
                              >
                                        Size
                              </th>
                              <th />
                    </tr>
          </thead>
          <tbody>
                    
          </tbody>
</Table>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown
    items={Array []}
/>,
}
`;
